### PR TITLE
Tweet model

### PR DESCRIPTION
**Pull request by lukaspode**
https://trello.com/c/R1L1Dn4n

### CODE REVIEW

File: spec/models/tweet_spec.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Tweet, type: :model do
```
Comment (martin8767) [inline spec/models/tweet_spec.rb:5]: add another describe testing the associations

File: app/models/user.rb [hunk 2]
```diff
@@ -4,6 +4,8 @@ class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
 
+  has_many :tweets, dependent: :destroy
```
Comment (martin8767) [inline app/models/user.rb:4]: add the association spec on the user_spec.rb

File: db/migrate/20231201174123_create_tweets.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class CreateTweets < ActiveRecord::Migration[7.1]
+  def change
+    create_table :tweets do |t|
+      t.text :content
```
Comment (martin8767) [inline db/migrate/20231201174123_create_tweets.rb]: add null false to the content

File: db/migrate/20231201174123_create_tweets.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class CreateTweets < ActiveRecord::Migration[7.1]
+  def change
+    create_table :tweets do |t|
+      t.text :content, null: false
+      t.references :user, null: false, foreign_key: true
```
Comment (micaelalorenzo) [inline db/migrate/20231201174123_create_tweets.rb:7]: Fun fact (no need to commit this change): You can also use the alias `belongs_to` to create this reference (it's the same method but it’s even clearer to read)

File: spec/models/tweet_spec.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  describe 'validations' do
+    it { is_expected.to validate_presence_of(:content) }
+    it { is_expected.to validate_length_of(:content).is_at_least(1).is_at_most(280) }
+  end
+
+  describe 'associations' do
+    it { should belong_to(:user) }
```
Comment (micaelalorenzo) [inline spec/models/tweet_spec.rb]:
```suggestion
    it { is_expected.to belong_to(:user) }
```
You can take a look at `git commit --ammend --no-edit` to push these changes.

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -47,4 +47,8 @@
       end
     end
   end
+
+  describe 'associations' do
+    it { should have_many(:tweets) }
```
Comment (micaelalorenzo) [inline spec/models/user_spec.rb]:
```suggestion
    it { is_expected.to have_many(:tweets) }
```

File: spec/factories/tweets.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :tweet do
+    content { Faker::Lorem.paragraph_by_chars(number: Faker::Number.between(from: 1, to: 160)) }
```
Comment (micaelalorenzo) [inline spec/factories/tweets.rb]:
```suggestion
    content { Faker::Lorem.paragraph_by_chars(number: Faker::Number.between(from: 1, to: 280)) }
```

File: spec/models/user_spec.rb (hunk 6 above)
Comment (micaelalorenzo) [reply spec/models/user_spec.rb]: I think you forgot this one

**Review by martin8767 — APPROVED** [review]

**Pull request merged**